libs/ngxtension/linked-query-param/src/linked-query-param.ts

```diff
@@ -0,0 +1,414 @@
+				this._navigationExtras = {};
+				return value;
+			});
+	}
+}
+
+type LinkedQueryParamOptions = {
+	/**
+	 * The injector to use to inject the router and activated route.
+	 */
+	injector?: Injector;
+} & Partial<NavigateMethodFields>;
+
+/**
+ * These are the function types that will be used to parse and serialize the query param value.
+ */
+type ParseFn<T> = (value: string | null) => T;
+type SerializeFn<T> = (value: T) => string | number | null;
+
+/**
+ *These types will be used to define the return types of the `set` and `update` methods of the signal.
+ * We need to re-type the WritableSignal, so that the set and update methods can have null in the call signature.
+ * But the WritableSignal itself won't have null in the call signature, so we need to re-type it.
+ * This is needed in order to be able to reset the value to null,
+ * which is not possible with the WritableSignal that doesn't have null in it's type.
+ */
+type SignalSetFn<T> = (value: T) => void;
+type SignalUpdateFn<T> = (fn: (value: T) => T) => void;
+
+/**
+ * Creates a signal that is linked to a query parameter.
+ *
+ * You can parse the query param value before it is passed to the signal, this way you can transform the value from a string to a number or boolean or whatever you need.
+ * You can also serialize the value before it is passed to the query param, this way you can serialize the value from a number or boolean or object to a string or null.
+ *
+ * You can also use the `defaultValue` option to set a default value if the query param is not present in the url (null or undefined).
+ * NOTE: You cannot use both `defaultValue` and `parse` at the same time. You should use `parse` instead to handle the default value.
+ *
+ * You can set the signal to update the query parameter by calling the `set` or `update` method.
+ * Both methods will accept the value + null as a valid value, so you can remove the query parameter by passing null if needed.
+ *
+ * The 'set' and 'update' methods will update the value synchronously, but will schedule the navigation event to
+ * happen on the next tick (using root effect scheduling). This means the query params will be updated asynchronously.
+ * The changes will be coalesced into a single navigation event. This means that if you call `set` or `update` multiple times
+ * in a row (synchronously), only the last value will be updated in the query params.
+ *
+ * If you have multiple signals listening to the same query parameter, they will all be updated when the navigation event happens.
+ *
+ * @param key The name of the query parameter.
+ * @param options Configuration options for the signal.
+ * @returns A signal that is linked to the query parameter.
+ */
+export function linkedQueryParam<T = string>(
+	key: string,
+	options: LinkedQueryParamOptions & {
+		parse: ParseFn<T>;
+		serialize: SerializeFn<T>;
+	},
+): WritableSignal<T> & {
+	set: SignalSetFn<T | null>;
+	update: SignalUpdateFn<T | null>;
+};
```

Comment:
nit/suggestion: I would make it explicitly clear that we'll be overriding `set` and `update` to allow `null`

```suggestion
): Omit<WritableSignal<T>, 'set' | 'update'> & {
	set: SignalSetFn<T | null>;
	update: SignalUpdateFn<T | null>;
};
```